doc/source/contributing/sphinx_doc_build.md

```diff
@@ -0,0 +1,46 @@
+(sphinx_doc_build)=
+# Building the documentation
```

Comment:
this document is great, I like how the sections have a couple sentence intro plus link to extended doc (which will be shared between code and doc contributions so we don't duplicate too much). The github preview section is also great, it is only missing the recommended local build instructions that I can add once we merge.